attrs_bsd.go

```diff
@@ -0,0 +1,44 @@
+// +build darwin freebsd netbsd
+// +build cgo
+
+package sftp
+
+import (
+	"os"
+	"syscall"
+
+	sshfx "github.com/pkg/sftp/internal/encoding/ssh/filexfer"
+)
+
+func attributesFromFileInfo(fi os.FileInfo) sshfx.Attributes {
+	var attrs sshfx.Attributes
+
+	switch sys := fi.Sys().(type) {
+	case *syscall.Stat_t:
+		attrs.SetSize(uint64(sys.Size))
+		attrs.SetUIDGID(sys.Uid, sys.Gid)
+		attrs.SetPermissions(sshfx.FileMode(sys.Mode))
+		attrs.SetACModTime(uint32(sys.Atimespec.Sec), uint32(sys.Mtimespec.Sec))
+
+	case *FileStat:
+		attrs.SetSize(sys.Size)
```

Comment:
It occurs to me that I’m being quite silly about copying these values over. I wrote a `FileStat.toAttributes()`, and when they’re both Attributes, I can just shallow copy, then copy the extension data over… 🤦